openshift_tools/monitoring/generic_metric_sender.py

```diff
@@ -0,0 +1,45 @@
+#!/usr/bin/env python
+# vim: expandtab:tabstop=4:shiftwidth=4
+"""
+Base class for sending metrics
+"""
+
+import os
+import yaml
+
+
+class GenericMetricSenderException(Exception):
+    '''
+        GenericMetricSenderException
+        Exists to propagate errors up from the api
+    '''
+    pass
+
+
+class GenericMetricSender(object):
+    """
+    Abstract class that encapsulates all metric sending operations.
+    """
+
+    def __init__(self):
+        self.config = None
+        self.config_file = None
+
+    def parse_config(self):
+        """ parse default config file """
+        if not self.config:
+            if not os.path.exists(self.config_file):
+                raise GenericMetricSenderException(self.config_file + " does not exist.")
+            self.config = yaml.load(file(self.config_file))
+
+    def add_dynamic_metric(self, discovery_key, macro_string, macro_array, host=None, synthetic=False):
+        pass
```

Comment:
I see what you're going for, and that's fine.

However, we DO need dynamic metric supported by the hawkular calls as well. Dynamic metrics are anything where the script needs to append a run time specific value.

So, for instance, our disk space check. When we write it, we have no idea what devices the host will have. So we write it as a dynamic metric and then pass into the dynamic metric /dev/xvda or /dev/xvdb, etc to it.

In hawkular, what I envision is that we'll simply add more tags for the dynamic items.

I think this will make more sense once you try to convert a collector script that uses dynamic items over to work with hawkular as well.